js/ui/grid_core/ui.grid_core.header_filter_core.js

```diff
@@ -168,10 +159,14 @@ export const HeaderFilterView = modules.View.inherit({
         }
     },
 
-    _getSearchExpr: function(options) {
+    _getSearchExpr: function(options, headerFilterOptions) {
         const lookup = options.lookup;
         const useDefaultSearchExpr = options.useDefaultSearchExpr;
-        const headerFilterDataSource = options.headerFilter && options.headerFilter.dataSource;
+        const headerFilterDataSource = headerFilterOptions.dataSource;
+
+        if(headerFilterOptions.search.searchExpr) {
+            return headerFilterOptions.search.searchExpr;
+        }
```

Comment:
Minor comment

Let's make it shorter:
```suggestion
        const headerFilterDataSource = headerFilterOptions.dataSource;
        const filterSearchExpr = headerFilterOptions.search.searchExpr;

        if(filterSearchExpr) {
            return filterSearchExpr;
        }
```